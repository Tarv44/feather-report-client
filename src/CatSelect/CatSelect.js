import React, {Component} from 'react';
import styles from './CatSelect.module.css';
import config from '../config';
import ProductContext from '../productContext';


export default class CatSelect extends Component {
    static contextType = ProductContext

    state = {
        newCategory: '',
        creatingNew: false
    }

    updateCreatingNew = (e) => {
        e.preventDefault();
        this.setState({ creatingNew: true })
    }

    updateCat = (newCategory) => {
        this.setState({ newCategory })
    }

    handleSubmit = (e) => {
        e.preventDefault()

        const headers = {
            'content-type': 'application/json'
        }

        const body = {
            title: this.state.newCategory,
            company: this.context.company.id
        }

        const options = {
            method: 'POST',
            headers,
            body: JSON.stringify(body)
        }

        fetch(`${config.API_ENDPOINT}/categories/`, options)
            .then(res => {
                if (!res.ok) {
                    return res.json().then(err => { throw err })
                }
                return res.json()
            })
            .then(feat => {
                this.props.handleNewCat(feat, this.props.index)
                this.setState({ creatingNew: false })
            })
    }

    render() {
        return this.state.creatingNew
        ? (
            <div className={styles.category}>
                <input 
                    onChange={e => this.updateCat(e.target.value, this.props.index)} 
                    type='text' 
                />
                <button 
                    className={styles.button} 
                    onClick={e => this.handleSubmit(e)}
                >
                    Submit Feature
                </button>
            </div>
        )
        : (
            <div className={styles.category}>
                <select 
                    onChange={e => this.props.handleCategory(e.target.value)}
                    value={this.props.category}
                >
                    <option>Select Category</option>
                    {this.props.catOptions}
                </select>
                <button
                    className={styles.button}
                    onClick={e => this.updateCreatingNew(e)}
                >
                    Create New Category
                </button>
            </div>
        )
    }
}