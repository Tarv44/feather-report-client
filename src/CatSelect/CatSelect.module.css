.category {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    border-bottom: solid black 1px;
    padding: .25em 0 .5em;
}

.button {
    margin-bottom: .25em;
}
