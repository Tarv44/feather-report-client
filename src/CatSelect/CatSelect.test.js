import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import CatSelect from './CatSelect'

describe('----- CatSelect -----', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(<CatSelect/>, div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(<CatSelect/>)
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})