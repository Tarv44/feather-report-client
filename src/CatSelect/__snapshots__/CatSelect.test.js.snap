// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- CatSelect ----- renders the UI as expected 1`] = `
<div
  className="category"
>
  <select
    onChange={[Function]}
  >
    <option>
      Select Category
    </option>
  </select>
  <button
    className="button"
    onClick={[Function]}
  >
    Create New Category
  </button>
</div>
`;
